### ISSUE TITLE
Inventory failing when flexible and uniform scalesets in the same RG

### ISSUE THREAD

**Issue by Justwmz** [ansible/ansible_collections_azure#1431]
##### SUMMARY
Inventory fails when flexible and uniform scale sets in the same RG.
I can see instances of the flexible scale set if the params "include_vmss_resource_groups" are not assigned.
I think it is because instances in the flexible scale set are created as standalone VMs, so Ansible is able to represent them in the inventory output.
But when I add back include_vmss_resource_groups, inventory is falling.
It seems to me like ansible is not able to parse a list of the instances from the flexible scale set since they have a different machine type.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
azure.azcollection

##### ANSIBLE VERSION
2.16.0

##### COLLECTION VERSION
2.1.1

##### STEPS TO REPRODUCE
Create two types of scale sets in the same RG.
In the inventory assign the params below:
include_vm_resource_groups:
 `- name of the RG`
include_vmss_resource_groups:
 `- name of the RG`

##### EXPECTED RESULTS
Inventory output should present instances from both types of scale sets.

**Comment by Fred-sun** [comment]
@Justwmz Obtaining the unifrom VMSS instance fails because the attribute ids of flexible VMSS and uniform VMSS instances are different. Procedure I fixed this in P #1451. In addition, if you want to obtain uniform vmss instances, set the parameter 'include_vmss_resource_groups' in your dynamic inventory scripts. Thank you!
```e.g.
/subscriptions/xxxxxxx/resourceGroups/v-xisuRG-uniform/providers/Microsoft.Compute/virtualMachineScaleSets/uniform-vmss/virtualMachines/0
/subscriptions/xxxxxxx/resourceGroups/v-xisuRG-flexible/providers/Microsoft.Compute/virtualMachines/flexible-vmss_6b8cb26f
```
sample as:
```e.g.
plugin: azure.azcollection.azure_rm
include_vmss_resource_groups:
    - ****-uniform
    - ****-flexible
******************
auth_source: auto

**Comment by nskidotgo** [comment]
I am also running into this issue, we have both flexible and uniform scalesets in the same RG. Any workarounds? thanks
